Data/Complex/Polar.hs

```diff
@@ -103,47 +151,48 @@ cis theta = mkPolar 1 theta
 polar :: (RealFloat a) => Polar a -> (a,a)
 {-# SPECIALISE polar :: Polar Double -> (Double,Double) #-}
 {-# SPECIALISE polar :: Polar Float  -> (Float,Float)   #-}
-polar (r:<theta) = (r,theta)
+polar (r :<< theta) = (r,theta)
 
 -- | The nonnegative magnitude of a complex number.
 magnitude :: (RealFloat a) => Polar a -> a
 {-# SPECIALISE magnitude :: Polar Float  -> Float  #-}
 {-# SPECIALISE magnitude :: Polar Double -> Double #-}
-magnitude (r:<_) = r
+magnitude (r :<< _) = r
 
 -- | The phase of a complex number, in the range @(-'pi', 'pi']@.
 -- If the magnitude is zero, then so is the phase.
 phase :: (RealFloat a) => Polar a -> a
 {-# SPECIALISE phase :: Polar Float  -> Float  #-}
 {-# SPECIALISE phase :: Polar Double -> Double #-}
-phase (_:<theta) = theta
+phase (_ :<< theta) = theta
 
 instance (RealFloat a) => Num (Polar a) where
     {-# SPECIALISE instance Num (Polar Float)  #-}
     {-# SPECIALISE instance Num (Polar Double) #-}
-    z + z'      = fromComplex (fromPolar z + fromPolar z')
-    z - z'      = fromComplex (fromPolar z - fromPolar z')
-    z * z'      = mkPolar  (magnitude z * magnitude z') (phase z + phase z')
-    negate z    = mkPolar_ (negate (magnitude z)) (phase z)
-    abs z       = mkPolar_ (magnitude z) 0
-    signum z    = mkPolar_ 1 (phase z)
-    fromInteger = flip mkPolar 0 . fromInteger
+    z + z'           = fromComplex (fromPolar z + fromPolar z')
+    z - z'           = fromComplex (fromPolar z - fromPolar z')
+    z * z'           = mkPolar  (magnitude z * magnitude z') (phase z + phase z')
+    negate z         = mkPolar  (negate (magnitude z)) (phase z)
+    abs z            = mkPolar_ (magnitude z) 0
+    signum (0 :<< _) = 0
+    signum z         = mkPolar_ 1 (phase z)
+    fromInteger      = flip mkPolar 0 . fromInteger
```

Comment:
The rest of the this instance is unchanged. The diff touches every line because of the formatting.